### PR TITLE
Add a function that the player can create username in the splash screen

### PR DESCRIPTION
**Pull request by cll69**
The dungeonView also can show the name.

Add the LoseOneLifeScreen, so that the dungeonView can get the username again when the player lost one life and restart the game. 

The game also can get username when the player chose to Try Again.

### CODE REVIEW

File: src/main/java/uk/ac/bath/se/SplashScreen.java [hunk 1]
```diff
@@ -7,6 +7,19 @@
 
 class SplashScreen {
 
+    //this stArea is for saving the user name created by the player and past it to the DungeonView.
+    // so the game can save the historical records
+
+    static String stArea;
```
Comment (theaaron123) [inline src/main/java/uk/ac/bath/se/SplashScreen.java]: I don't think this should be static as it has no default value

File: src/main/java/uk/ac/bath/se/SplashScreen.java [hunk 2]
```diff
@@ -7,6 +7,19 @@
...
+
+    public void setstArea(String stArea) {
+        this.stArea = stArea;
+    }
+
+    public static String getstArea(String stArea) {
```
Comment (theaaron123) [inline src/main/java/uk/ac/bath/se/SplashScreen.java]: the object is not instantiated the stArea string has no value so static not needed

File: src/main/java/uk/ac/bath/se/DungeonView.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
     private boolean collided = false;
     Dungeon dungeon;
 
-    public DungeonView() {
+    public DungeonView(String string) {
```
Comment (theaaron123) [inline src/main/java/uk/ac/bath/se/DungeonView.java]: I'm not sure about passing it in this way but for now could you rename it playerName or something more descriptive please

**Review by theaaron123 — CHANGES_REQUESTED** [review]
A couple of requests

File: src/main/java/uk/ac/bath/se/DungeonView.java (hunk 3 above)
Comment (cll69) [reply src/main/java/uk/ac/bath/se/DungeonView.java]: Yeah. I can do it.

File: src/main/java/uk/ac/bath/se/SplashScreen.java (hunk 1 above)
Comment (cll69) [reply src/main/java/uk/ac/bath/se/SplashScreen.java]: I tried to make it no static in the first time, but the dungeon cannot accept it.

File: src/main/java/uk/ac/bath/se/SplashScreen.java (hunk 2 above)
Comment (cll69) [reply src/main/java/uk/ac/bath/se/SplashScreen.java]: Yeah. I can do it.

**Pull request merged**